### ISSUE TITLE
Reduce code duplication

### ISSUE THREAD

**Issue by jan-christiansen** [sagr4019/ResearchProject#90]
Move functions that are defined multiple times in the files `train_keras.py`, `train.py`, and `train_seq_embed.py` like `split_dataset` and `shuffle` into separate file.

**Comment by DanielEggelmann** [comment]
Duplicate function have been moved to the file `lib.py`.

**Comment by DanielEggelmann** [comment]
At the moment we have three implementations with the more or less same functionality. Therefore I would suggest to get rid of two implementations (`train.py, train_keras.py, compile.py, asm.py, preprocess_data.py, preprocess_data_string.py`) and only keep the sequence embedding implementation, which seems to me the best one.
